### ISSUE TITLE
The result of TextIOWrapper.detach() should be IO[bytes]

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/typeshed#3993]
For the following example:
```python
import sys, io

if isinstance(sys.stdout, io.TextIOWrapper) and sys.version_info < (3, 7):
    sys.stdout = io.TextIOWrapper(
        sys.stdout.detach(),
        encoding="ascii",
        errors="replace")
```
I get the following error:
```
reconfigure36.py:5: error: Argument 1 to "TextIOWrapper" has incompatible type "IOBase"; expected "IO[bytes]"
```
Should not `TextIOWrapper.detach()` return `IO[bytes]` instead of `IOBase`?

Replacing `sys.stdout.detach()` with `sys.stdout.buffer` fixes the complains.

**Comment by srittau** [comment]
IO types predate protocols and are a mess. `detach()` should ideally return whatever it returns in the implementation (I assume this is the first argument passed to its constructor?) and `TextIOWrapper` should ideally expect a custom protocol and things should work. That said, your suggestion would most likely work fine for now. PR welcome!
